### PR TITLE
Add badge and soundName properties to local notification schedule

### PR DESCRIPTION
**Pull request by ogtfaber**
I'm using local notification with badges on iOS, here's a pull request for support. I've sticked to the same syntax. The PushNotificationIOS library expects an integer for the applicationIconBadgeNumber.

### CODE REVIEW

**Comment by npomfret** [comment]
Hi - apologies for the delay, and thank you for submitting this change.  Struggling to find time to look at PR's on this project. 

Can you please make sure you describe the motivation for this change, and crucially **what is the expected change in behaviour**?  How could someone test it?

I would suggest you ask the authors of other outstanding PR's to test your change in a private fork.  If you can get the thumbs up from a couple of other developers then I'd be happy to merge it.  In exchange you could offer to test theirs maybe..?  That would really help get this project moving.

**Comment by mcmatan** [comment]
@ogtfaber hi there, I have an open pull request too, and I'm offering to test yours If you test mine (:

This is my PR:

https://github.com/zo0r/react-native-push-notification/pull/340

If I'ts ok for you, please specify what are the ways of testing (:
Cheers

**Comment by ogtfaber** [comment]
@npomfret Appreciate your work for this package, thanks! 

The expected behavior of the change is that the features the remote notifications support are also working for the local notifications. As far as I can see the feature @mcmatan did a PR for is also in my fork, as well as the badge number. @mcmatan can you give a thumbs up on my pull request?

**Comment by mcmatan** [comment]
@npomfret  You got it!

**Comment by ogtfaber** [comment]
@mcmatan did you test my PR? If so we can give @npomfret a go :)

**Comment by mcmatan** [comment]
@ogtfaber Yes I did

File: index.js [hunk 1]
```diff
@@ -155,9 +155,18 @@ Notifications.localNotification = function(details: Object) {
...
+
+		if (details.hasOwnProperty('playSound') && !details.playSound) {
+			soundName = ''; // empty string results in no sound (and no vibration)
+		}
+
 		this.handler.scheduleLocalNotification({
 			fireDate: details.date.toISOString(),
 			alertBody: details.message,
+			soundName: soundName,
+			applicationIconBadgeNumber: parseInt(details.number, 10),
```
Comment (npomfret) [inline index.js]: What's the `10` for here?

**Comment by npomfret** [comment]
I've added a comment, could someone please reply?  

Are there any documentation changes needed?

@ogtfaber are you satisfied that:

 * this change works as described?
 * the purpose of this change is appropriate for the project?
 * the code is _sensibly_ written (in your opinion)?

File: index.js (hunk 1 above)
Comment (gowza) [inline index.js]: To indicate that the string input number to parse is of base 10.

10 is default anyway, but people like to be explicit.

**Comment by npomfret** [comment]
@mcmatan are you happy with this PR? 

@ogtfaber if we don't here from @mcmatan can you get someone else to take a look.. Refer them to [this](https://github.com/zo0r/react-native-push-notification/blob/master/submitting-a-pull-request.md).

**Comment by mcmatan** [comment]
@npomfret I'm sorry for my late response.

-Are there any documentation changes needed?
No. 

-this change works as described?
Yes, sound is playing and badge is displayed.

-the purpose of this change is appropriate for the project?
Yes, it only adds/fixes functionality

-the code is sensibly written (in your opinion)?
Yes.

**Comment by dpikt** [comment]
Wow, I needed these changes for a project I'm working on. Great timing y'all 😀

**Comment by ogtfaber** [comment]
Sorry for the late reply, but thanks for merging @npomfret !

**Pull request merged**